lib/features/campaigns/screens/mixins/search_mixin.dart

```diff
@@ -0,0 +1,251 @@
+                    enabledBorder: OutlineInputBorder(borderSide: BorderSide.none),
+                    focusedBorder: OutlineInputBorder(borderSide: BorderSide.none),
+                  ),
+                  style: theme.textTheme.labelMedium?.apply(color: ThemeColors.textDark),
+                  onChanged: (value) {
+                    setState(() {
+                      _campaignSettings.searchString = value;
+                    });
+                  },
+                  onFieldSubmitted: (value) => _searchAddress(),
+                ),
+              ),
+              _showClearSearch
+                  ? IconButton(
+                      onPressed: () {
+                        setState(() {
+                          _controller!.text = '';
+                          _campaignSettings.searchString = '';
+                          _searchResult!.clear();
+                        });
+                      },
+                      icon: Icon(
+                        Icons.close,
+                        color: ThemeColors.textDisabled,
+                      ),
+                    )
+                  : SizedBox(width: 24),
+            ],
+          ),
+        ),
+      );
+      widgets.add(searchBar);
+
+      var collapseButton = Positioned(
+        top: 5,
+        right: -6,
+        child: ElevatedButton(
+          onPressed: _hideSearchBar,
+          style: ElevatedButton.styleFrom(
+            shape: CircleBorder(),
+            padding: EdgeInsets.all(12),
+            backgroundColor: ThemeColors.background, // <-- Button color
+          ),
+          child: Icon(
+            Icons.keyboard_arrow_up,
+          ),
+        ),
+      );
+      widgets.add(collapseButton);
+    }
+
+    if (_showSearchBar && _searchResult!.isNotEmpty) {
+      var listWidgets = _searchResult!.map((item) => _getSearchResultWidget(item, theme)).toList();
+      for (var i = 0; i < listWidgets.length; i += 2) {
+        listWidgets.insert(
+          i,
+          SizedBox(height: 6),
+        );
+      }
+
+      var searchResultWidget = Positioned(
+        top: 66,
+        bottom: 6,
+        left: 6,
+        right: 6,
+        child: SingleChildScrollView(
+          child: Column(
+            children: [
+              ...listWidgets,
+            ],
+          ),
+        ),
+      );
+      widgets.add(searchResultWidget);
+    }
+
+    return widgets;
+  }
+
+  void _searchAddress() async {
+    FocusManager.instance.primaryFocus?.unfocus();
+    var scaffoldMessenger = ScaffoldMessenger.of(context);
+
+    var nominatimService = NominatimService();
+    var addressList =
+        await nominatimService.findStreetOrCity(_controller!.text.trim(), CampaignConstants.viewBoxGermany);
+    final added = <String>{};
+
+    final distinctAddresses = addressList.where((item) => added.add(item.displayName)).toList();
+    var currentMapLocation = _campaignSettings.lastPosition!;
+    distinctAddresses.sort((itemA, itemB) {
```

Comment:
Always confusing when sorting mutates on a const-like item 🙃  (but no need to change)